conda/gateways/logging.py

```diff
@@ -35,11 +35,25 @@ def filter(self, record):
         return True
 
 
+# Don't use initialize_logging/initialize_root_logger/initialize_conda_logger in
+# cli.python_api! There we want the user to have control over their logging,
+# e.g., using their own levels, handlers, formatters and propagation settings.
+
 @memoize
 def initialize_logging():
-    # initialize_root_logger()  # probably don't need to touch the root logger per #5356
+    # root and 'conda' logger both get their own separate sys.stderr stream handlers.
+    # root gets level ERROR; 'conda' gets level WARN and does not propagate to root.
+    initialize_root_logger()
     initialize_conda_logger()
+    initialize_std_loggers()
+
 
+@memoize
+def initialize_std_loggers():
+    # Set up special loggers 'stdout'/'stderr' which output directly to the corresponding
+    # sys streams, filter token urls and don't propagate.
+    # TODO: To avoid clashes with user loggers when cli.python_api is used, these loggers
+    #       should most likely be renamed to 'conda.stdout'/'conda.stderr' in the future!
```

Comment:
Agreed.  I'd thought of that as well.  I'm going to repoint this PR to 4.3.x.  So that we clean up the mess I made trying to get this logging stuff worked out in the first place.  However, I'd be fine renaming these loggers in 4.4.x.

Also, there are several "old" loggers in `conda/console.py`.  It's always been the intent to move the code that's still relevant into `conda/gateways/logging.py`.  Might make sense to do that as part of a logging cleanup PR for 4.4.x.